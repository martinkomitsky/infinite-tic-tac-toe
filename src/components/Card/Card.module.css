.card {
	width: 100%;
	box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.2), 0 1px 1px 0 rgba(0, 0, 0, 0.14),
		0 2px 1px -1px rgba(0, 0, 0, 0.12);
	background: white;
	padding: 16px;
	align-items: center;
	flex-direction: column;
	display: flex;
}

.cardContainer {
	width: 100%;
}
