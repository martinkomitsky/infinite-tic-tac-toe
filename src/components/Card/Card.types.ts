export interface ICardProps {
	children: React.ReactChild;
}
