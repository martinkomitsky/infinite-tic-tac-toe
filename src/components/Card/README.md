# Card

Компонент карточки с тенюшкой, принимает в себя `children`.
