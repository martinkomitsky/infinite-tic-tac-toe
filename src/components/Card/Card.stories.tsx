import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Card } from './Card';

const sampleText = `
	Kanye Omari West (/ˈkɑːnjeɪ/; born June 8, 1977) is an American
	rapper, singer, songwriter, record producer, entrepreneur, and
	fashion designer. His music has spanned a broad range of styles,
	incorporating an eclectic range of influences including soul,
	baroque pop, electro, indie rock, synth-pop, industrial, and
	gospel.
`;

const stories = storiesOf('Infinite Tic-Tac-Toe|Card', module).addParameters({
	readme: {
		sidebar: require('./README.md'),
	},
});

stories.addDecorator(withKnobs);

stories.add('with a child', () => (
	<React.Fragment>
		<Card>
			<React.Fragment>{text('Children text', sampleText)}</React.Fragment>
		</Card>
	</React.Fragment>
));
