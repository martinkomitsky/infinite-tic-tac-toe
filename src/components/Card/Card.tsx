import * as React from 'react';
import * as s from './Card.module.css';
import * as T from './Card.types';

export const Card: React.FunctionComponent<T.ICardProps> = props => {
	const { children } = props;

	return (
		<div className={s.card}>
			<div className={s.cardContainer}>{children}</div>
		</div>
	);
};
