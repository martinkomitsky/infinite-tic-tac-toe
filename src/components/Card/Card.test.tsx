import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { shallow } from '../../../tests/setup/test-setup';
import { Card } from './Card';

describe('<Card />', () => {
	let wrapper: any;
	const text = 'test sest';

	beforeEach(() => {
		wrapper = shallow(<Card>{text}</Card>);
	});

	it('should match the snapshot', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('renders children', () => {
		expect(wrapper.text()).toEqual(text);
	});
});
