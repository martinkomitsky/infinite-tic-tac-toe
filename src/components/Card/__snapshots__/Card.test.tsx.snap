// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match the snapshot 1`] = `
<div>
  <div>
    test sest
  </div>
</div>
`;
